geo/point.go

```diff
@@ -0,0 +1,125 @@
+package geo
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/paulmach/orb"
+	"github.com/paulmach/orb/geo"
+	"github.com/paulmach/orb/planar"
+	"go.starlark.net/starlark"
+)
+
+func newPoint(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (v starlark.Value, err error) {
+	var (
+		x, y     starlark.Value
+		lat, lng float64
+	)
+	v = starlark.None
+
+	if err = starlark.UnpackArgs("Point", args, kwargs, "x", &x, "y", &y); err != nil {
+		return
+	}
+
+	if lat, err = float64FromNumericValue(x); err != nil {
+		return
+	}
+	if lng, err = float64FromNumericValue(y); err != nil {
+		return
+	}
+
+	return Point{lat, lng}, nil
+}
+
+// Point is the starlark geographic point type
+type Point [2]float64
+
+// assert point is a starlark value
+var _ starlark.Value = (*Point)(nil)
+
+// String implements the starlark.Value interface
+func (p Point) String() string { return fmt.Sprintf("(%f,%f)", p[0], p[1]) }
+
+// Type implements the starlark.Value interface
+func (p Point) Type() string { return "Point" }
+
+// Freeze implements the starlark.Value interface, point is immutable
+func (p Point) Freeze() {}
+
+// Truth implements the starlark.Value interface
+func (p Point) Truth() starlark.Bool {
+	return starlark.Bool(p[0] != 0 && p[1] != 0)
+}
+
+// Hash implements the starlark.Value interface
+func (p Point) Hash() (uint32, error) {
+	x, _ := floatHash(p[0])
+	y, _ := floatHash(p[1])
+	// TODO (b5): this is bad. fix
+	return x - y, nil
+}
+
+// Attr gets an attribute of point
+func (p Point) Attr(name string) (starlark.Value, error) {
+	switch name {
+	case "x", "lat":
+		return starlark.Float(p[0]), nil
+	case "y", "lng":
+		return starlark.Float(p[1]), nil
+	case "buffer":
+		return addClosure("buffer", p, p.buffer)
+	case "distance":
+		return addClosure("distance", p, p.distance)
+	case "distanceGeodesic":
+		return addClosure("distanceGeodesic", p, p.distanceGeodesic)
+	case "KNN":
```

Comment:
Would it be better to rename this KNearestNeighbor? I'm not aware of whether or not "KNN" is immediately obvious to people in the geo data world.

Reply:
as per Shapely/our RFC apparently it is 🤷‍♀️

Reply:
Ok! As long as there's a reason.